frontend/src/components/common/podcast-filter/podcast-filter.tsx

```diff
@@ -0,0 +1,63 @@
+import { Path, FieldValues } from 'react-hook-form';
+import { Genre, GenresFilter, PodcastLoadFilter } from 'common/types/types';
+import { ButtonType, ButtonColor } from 'common/enums/enums';
+import { useAppForm } from 'hooks/hooks';
+import { Button, Checkbox } from 'components/common/common';
+import styles from './styles.module.scss';
+
+type Props = {
+  genres: Genre[];
+  onApply: (data: GenresFilter) => void;
+  onCancel: () => void;
+  currentState: PodcastLoadFilter;
+};
+
+const PodcastFilter: React.FC<Props> = ({
+  genres,
+  onApply,
+  onCancel,
+  currentState,
+}) => {
+
+  const currentValues = genres.reduce<Record<string, boolean>>((State, genre) => {
+    State[genre.key] = currentState.genres.includes(genre.id);
+    return State;
+  }, {});
```

Comment:
```suggestion
  const currentValues = genres.reduce<Record<string, boolean>>((acc, genre) => {
    return {
      ...acc,
      [genre.key] = currentState.genres.includes(genre.id);
    };
  }, {});
```
create a helper from it and move to frontend/src/components/common/podcast-filter/helpers/